src/app-logic/browser-connection.js

```diff
@@ -108,13 +103,8 @@ class BrowserConnectionImpl implements BrowserConnection {
     }
     // For older versions, fall back to the geckoProfiler frame script API.
     // This fallback can be removed once the oldest supported Firefox ESR version is 96 or newer.
-    if (this._geckoProfiler) {
-      return this._geckoProfiler.getSymbolTable(debugName, breakpadId);
-    }
-
-    throw new Error(
-      'Cannot obtain a symbol table: have neither WebChannel nor a GeckoProfiler object'
-    );
+    const geckoProfiler = await this._getConnectionViaFrameScript();
```

Comment:
Hmm, this has no timeout. Wouldn't this mean that you get a hang in the following case:

 1. Be on a version of Firefox which does not support the new WebChannel.
 2. Open a fresh tab and go to profiler.firefox.com
 3. Drop an unsymbolicated profile into the profiler

We will create  the browserConnection (because the (old) WebChannel exists) but there's no frame script, so the geckoProfilerPromise will never resolve.

Reply:
I just tried and you're right, I'll resurrect some of the old code to fix this.